server/src/main/java/org/candlepin/policy/js/autobind/AutobindRules.java

```diff
@@ -68,29 +67,38 @@ public AutobindRules(JsRunner jsRules, ProductCurator productCurator, RulesObjec
         List<Pool> pools, ComplianceStatus compliance, String serviceLevelOverride,
         Set<String> exemptLevels, boolean considerDerived) {
 
+        List<PoolQuantity> bestPools = new ArrayList<PoolQuantity>();
         int poolsBeforeContentFilter = pools.size();
         pools = filterPoolsForV1Certificates(consumer, pools);
         log.debug("pools.size() before V1 certificate filter: {}, after: {}",
             poolsBeforeContentFilter, pools.size());
 
+        List<String> productIdList;
+        if (productIds == null) {
+            productIdList = new ArrayList<String>();
+        }
+        else {
+            productIdList = Arrays.asList(productIds);
+        }
+        List<String> consumerInstalledProducts = new ArrayList<String>();
```

Comment:
We don't actually use this until we hit the logging bits on failure. I assume you're going for the whole DRY approach here, but I'd prefer to omit this unless necessary. I.E. Gate the whole thing in a log-level check like the example below.